### PR TITLE
Add TexLive

### PR DESCRIPTION
**Pull request by naveen521kk**
Bring in LaTeX. My plan is to follow Archlinux. First, this PR will add `texlive-bin` package, which in itself isn't useful. Then `texlive-core` must be added, I have no idea why Archlinux uses `"https://sources.archlinux.org/other/texlive/$pkgname-$pkgver-src.zip"` https://github.com/archlinux/svntogit-packages/blob/packages/texlive-core/trunk/PKGBUILD#L32 as a source, need to look into that. Later every collection can be added as a new package.

### CODE REVIEW

**Comment by lazka** [comment]
nice!

**Comment by Biswa96** [comment]
Isn't it possible to build the programs here instead of using prebuilt binaries?

**Comment by Biswa96** [comment]
Why is there -bin in the package name?

**Comment by naveen521kk** [comment]
> Why is there -bin in the package name?

As I told I am following Archlinux here. They have the core binaries in `texlive-bin`.

**Comment by naveen521kk** [comment]
I made https://github.com/msys2/msys2-texlive so that `texlive-core` can be built. I am slightly confused about the other setup's like configuration and how it is expected to be on Windows and I have contacted their [mailing list](https://tug.org/pipermail/tldistro/2021q2/000387.html) they could help me.

**Comment by naveen521kk** [comment]
Looks like it's going to be *hard* to get things right, I don't know much about configuration, which we can copy from Arch if they had texlive 2021, maybe let's wait for that to happen.

File: mingw-w64-texlive-bin/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+   cd Work
+   echo "--> Initial configuration..."
+   # we use temporary prefix to avoid messing the existing
+   # $pkgdir/usr/share/texmf tree
+   ../configure --prefix="${MINGW_PREFIX}" -C \
+     --build=${MINGW_CHOST} \
+     --host=${MINGW_CHOST} \
+     --target=${MINGW_CHOST} \
+     --disable-native-texlive-build \
+     --with-banner-add="/MSYS2 MINGW" \
```
Comment (naveen521kk) [inline mingw-w64-texlive-bin/PKGBUILD:72]: @lazka what do you think this banner should be?
It will be shown each time when launching latex.

Comment (lazka) [reply mingw-w64-texlive-bin/PKGBUILD:72]: no idea..

Comment (naveen521kk) [reply mingw-w64-texlive-bin/PKGBUILD:72]: like how we have something in gcc? can I copy the same?

Comment (lazka) [reply mingw-w64-texlive-bin/PKGBUILD:72]: sure

**Comment by naveen521kk** [comment]
If the checks passed or it build sucessfully this should be ready for testing.

**Comment by naveen521kk** [comment]
for ucrt build, Perl is missing no idea what to do.

**Comment by naveen521kk** [comment]
Looks like this requires quite some Perl modules for texlive utilities to work. I found https://tug.org/svn/texlive/trunk/Master/source/tlperl/tlperl.README?view=markup that needs to be added first.

**Comment by naveen521kk** [comment]
After adding those Perl modules this should be ready

**Comment by naveen521kk** [comment]
This is pretty much ready, AFAIK. I will check for the CI run I triggered in https://github.com/msys2/MINGW-packages/pull/8271/commits/e719cc73345a0776f7a168c304ded4c43a55074f and then do a local test, finally squash some commits so that this can be merged.

**Comment by naveen521kk** [comment]
Tested this PR locally, by downloading the artefacts and things are working, so time to clean.
TODO:
- [ ] Reduce output of the hooks
- [ ] Cleans things in PKGBUILD
- [ ] Squash some commits

**Comment by naveen521kk** [comment]
@lazka this works locally, review and merge(please squash while merging.)

**Comment by naveen521kk** [comment]
Thanks.

**Comment by podsvirov** [comment]
@naveen521kk, thanks for working to make this possible!

**Pull request merged**